mlflow/environment_variables.py

```diff
@@ -511,3 +511,8 @@ def get(self):
 MLFLOW_GATEWAY_RATE_LIMITS_STORAGE_URI = _EnvironmentVariable(
     "MLFLOW_GATEWAY_RATE_LIMITS_STORAGE_URI", str, None
 )
+
+#: If True, MLflow fluent logging APIs, e.g., `mlflow.log_metric` will log synchronously.
+MLFLOW_ENABLE_SYNCHRONOUS_LOGGING = _BooleanEnvironmentVariable(
+    "MLFLOW_ENABLE_SYNCHRONOUS_LOGGING", True
+)
```

Comment:
Can we make this `ENABLE_ASYNC_LOGGING`, since the default behavior is already synchronous logging & synchronous logging isn't something the user has to "enable"?

Reply:
in the future (actually near future IMO) we are going to turn on async by default, then `MLFLOW_ENABLE_SYNCHRONOUS_LOGGING` will be more straightforward. Also this is to keep consistency with our args, e..g, `mlflow.log_metric("foo", 1, synchronous=False)`.

Reply:
discussed offline, we will go with `MLFLOW_ASYNCHRONOUS_LOGGING `